src/acquisition/covidcast/covidcast_row.py

```diff
@@ -0,0 +1,270 @@
+from dataclasses import asdict, dataclass, field, fields
+from datetime import date
+from typing import Any, ClassVar, Dict, Iterable, List, Optional
+
+import pandas as pd
+from delphi_utils import Nans
+
+from delphi.epidata.acquisition.covidcast.csv_importer import CsvImporter
+from delphi.epidata.server.utils.dates import day_to_time_value, time_value_to_day
+from delphi.epidata.server.endpoints.covidcast_utils.model import PANDAS_DTYPES
+
+
+@dataclass
+class CovidcastRow:
+    """A container for the values of a single covidcast database row.
+
+    Used for:
+    - inserting rows into the database
+    - creating test rows with default fields for testing
+    - converting from and to formats (dict, csv, df, kwargs)
+    - creating consistent views, with consistent data types (dict, csv, df)
+
+    The rows are specified in 'v4_schema.sql'. The datatypes are made to match database. When writing to Pandas, the dtypes match the JIT model.py schema.
+    """
+
+    # Arguments.
+    source: str = "src"
+    signal: str = "sig"
+    time_type: str = "day"
+    geo_type: str = "county"
+    time_value: int = 20200202
+    geo_value: str = "01234"
+    value: float = 10.0
+    stderr: float = 10.0
+    sample_size: float = 10.0
+    missing_value: int = Nans.NOT_MISSING.value
+    missing_stderr: int = Nans.NOT_MISSING.value
+    missing_sample_size: int = Nans.NOT_MISSING.value
+    issue: Optional[int] = 20200202
+    lag: Optional[int] = 0
+    id: Optional[int] = None
+    direction: Optional[int] = None
+    direction_updated_timestamp: int = 0
+    value_updated_timestamp: int = 20200202
```

Comment:
I feel 😬 about including defaults here; I'm worried they will leak into the actual acquisition stream by mistake. The dedicated builder function (`_make_placeholder_row`) may be ugly, but it does enforce a clear line between what's live data and what's test data. With this, all it would take is for some future change to forget to set one of the 17 fields, and the default will go into the database instead.

Reply:
Decided to go with option 1 after talking to Katie.